src/Registry.ts

```diff
@@ -164,6 +169,11 @@ export class Registry extends Evented<{}, RegistryLabel, RegistryEventObject> im
 		this._widgetRegistry.set(label, promise);
 
 		promise.then((widgetCtor) => {
+
+			if (this._ctorIsDefaultExport<T>(widgetCtor)) {
```

Comment:
Why does this have to be on the registry instance? Could you have done something like this (not on instance)? 

```ts
export function isWidgetConstructorDefaultExport<T>(item: any): item is ESMDefaultWidgetBase <T> {
   return Boolean(
       module && 
       module.hasOwnProperty('__esModule') && 
       module.hasOwnProperty('default') && 
       isWidgetBaseConstructor(module.default)
   );
}
```

and used it like this (might have to use a different variable for the assignment):

```ts
if (isWidgetConstructorDefaultExport<T>(widgetCtor)) {
    widgetCtor = widgetCtor.default;
}
```

Reply:
Thanks @agubler was looking for assistance with these typings